06_puppy/kasunfdo/mapstore.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+func NewMapStore() *MapStore {
+	return &MapStore{count: 0, store: map[uint64]Puppy{}}
+}
+
+func (m *MapStore) CreatePuppy(puppy Puppy) uint64 {
+	m.count++
+	puppy.ID = m.count
+	m.store[puppy.ID] = puppy
+	return puppy.ID
+}
+
+func (m *MapStore) ReadPuppy(id uint64) (Puppy, bool) {
+	puppy, exists := m.store[id]
+	return puppy, exists
+}
+
+func (m *MapStore) UpdatePuppy(id uint64, puppy Puppy) bool {
```

Comment:
Better return an `error` - doesn't need to be a custom error type (that's part of the next lab I believe), but at least:
```go
fmt.Errorf("no puppy with ID %d", id)
```
To me it's nicer to only keep the ID encoded in `puppy`. It feels a bit dodgy to accept a puppy whose ID doesn't match the separate id field. You could add an error check for it, but really just use only one ID imo.  
Feel free to argue against it :)

Reply:
I did not use error as it has to be implemented in lab 7 😃
Anyway, I'll return error.